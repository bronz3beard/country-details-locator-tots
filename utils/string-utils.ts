export const normalizeString = (stringValue: string) => stringValue.trim().toLowerCase();
