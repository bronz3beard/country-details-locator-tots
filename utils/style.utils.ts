import { ClassValue, clsx } from 'clsx';
import { JSXElementConstructor } from 'react';
import { twMerge } from 'tailwind-merge';

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}

export function createNamespacedComponent<T extends JSXElementConstructor<any>, U>(
  getComponent: () => T,
  namespaceMembers: U
): T & U {
  const NamespaceComponent = getComponent();
  return Object.assign(NamespaceComponent, namespaceMembers);
}
