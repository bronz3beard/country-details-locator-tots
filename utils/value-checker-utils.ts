export function assertIsTrue(condition: boolean, errorMessage?: string): asserts condition {
  if (!condition) {
    throw new Error(errorMessage ?? 'Assertion failed');
  }
}

export const isFunction = <T extends (...args: any[]) => any = (...args: any[]) => any>(
  func: unknown
): func is T => {
  return func != null && typeof func === 'function';
};
