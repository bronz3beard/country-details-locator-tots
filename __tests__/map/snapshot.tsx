/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';
import MapPage from '~/app/map/page';
import { assertSnapshot } from '../utils/assert-snapshot';

const mockData = [
  {
    country: {
      awsRegion: 'mock-east-1',
      capital: 'Mock City',
      code: 'MK',
      continent: { name: 'Mock' },
      currencies: ['M'],
      currency: 'M',
      emoji: '🏁',
      emojiU: 'U+MOCK',
      languages: [{ name: 'MOCK', native: 'Móck' }],
      name: 'Mock Test',
      native: 'Mock Tést',
      phone: '0',
      phones: ['0'],
      states: [{ name: 'Mock Town', code: 'MT' }],
      subdivisions: []
    }
  }
];

jest.mock('~/app/map/api/actions', () => ({
  getMapData: jest.fn(() =>
    Promise.resolve({
      data: mockData,
      errors: null,
      error: null,
      loading: 'false' // this is a string to keep a jest errro quiet but in the typescript it is a boolean
    })
  )
}));

jest.mock('~/app/map/components/map-page-container', () => 'MapPageContainer');

describe('Snapshot MapPage', () => {
  beforeEach(() => {
    jest.resetModules();
    jest.clearAllMocks();
  });

  it('renders Map Page unchanged', async () => {
    const jsx = await MapPage();

    render(jsx);
    assertSnapshot(jsx);
  });
});
