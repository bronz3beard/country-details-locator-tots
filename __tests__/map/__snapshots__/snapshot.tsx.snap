// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot MapPage renders Map Page unchanged 1`] = `
<MapPageContainer
  countries={
    [
      {
        "country": {
          "awsRegion": "mock-east-1",
          "capital": "Mock City",
          "code": "MK",
          "continent": {
            "name": "Mock",
          },
          "currencies": [
            "M",
          ],
          "currency": "M",
          "emoji": "🏁",
          "emojiU": "U+MOCK",
          "languages": [
            {
              "name": "MOCK",
              "native": "Móck",
            },
          ],
          "name": "Mock Test",
          "native": "Mock Tést",
          "phone": "0",
          "phones": [
            "0",
          ],
          "states": [
            {
              "code": "MT",
              "name": "Mock Town",
            },
          ],
          "subdivisions": [],
        },
      },
    ]
  }
  loading="false"
/>
`;
